### PR TITLE
Build and load bokeh compiled model cache

### PR DESCRIPTION
**Pull request by philippjfr**
This PR customizes setup.py in such a way that it compiles custom bokeh models and stores the compiled code alongside the original code. Additionally it registers a hook with bokeh which loads the cached models during bokeh's compilation step.

### CODE REVIEW

**Comment by philippjfr** [comment]
@ceball Could you have a brief look at my changes to the .travis.yml? nodejs is now a build requirement so I've had to go back to the pip_on_conda approach.

**Comment by ceball** [comment]
The "pip on conda" approach was experimental, for geoviews, where *users* also cannot install geoviews using pip alone without taking additional steps (because of some of its dependencies). In contrast, for panel, isn't it only *package builders* who require more than pip to get all the dependencies? Is that right? If that's right, I would suggest a different approach.

However, either way, the "pip on conda" approach should still work (it's just not ideal), so you could carry on and I can fix up later. (I can't work right now.)

File: .travis.yml [hunk 1]
```diff
@@ -11,11 +11,11 @@ sudo: false
 env:
   global:
     - PYENV_VERSION=3.6
-    - CHANS_DEV="-c pyviz/label/dev -c jbednar"
+    - CHANS_DEV="-c pyviz/label/dev -c jbednar -c bokeh/label/dev"
```
Comment (ceball) [inline .travis.yml]: Just to check: is this addition a build time channel requirement, or do users need it too?

File: .travis.yml [hunk 2]
```diff
@@ -73,15 +73,16 @@ jobs:
         - doit package_build $CHANS $PKG_TEST_PYTHON --test-group=unit
       script: doit package_upload --token=$ANACONDA_TOKEN $LABELS
 
-
     - &pip_pkg
-      env: PYPI=testpypi PYPIUSER=$TPPU PYPIPASS=$TPPP
+      <<: *conda_default
+      env: PYPI="https://test.pypi.org/legacy/" PYPIUSER=$TPPU PYPIPASS=$TPPP
```
Comment (ceball) [inline .travis.yml]: Presumably reverting the shortcut was unintended side effect of copy/paste?

```suggestion
      env: PYPI=testpypi PYPIUSER=$TPPU PYPIPASS=$TPPP
```

I've never tried this 'suggestion' thing before...

File: .travis.yml [hunk 3]
```diff
@@ -90,7 +91,7 @@ jobs:
       env: DESC="" TRAVIS_NOCACHE=$TRAVIS_JOB_ID LABELS=$LABELS_REL CHANS=$CHANS_REL
 
     - <<: *pip_pkg
-      env: PYPI=pypi PYPIUSER=$PPU PYPIPASS=$PPP
+      env: PYPI="https://upload.pypi.org/legacy/" PYPIUSER=$PPU PYPIPASS=$PPP
```
Comment (ceball) [inline .travis.yml]:
```suggestion
      env: PYPI=pypi PYPIUSER=$PPU PYPIPASS=$PPP
```

File: tox.ini [hunk 4]
```diff
@@ -4,7 +4,6 @@
 [tox]
 #          python version             test group                  extra envs  extra commands       
 envlist = {py27,py35,py36}-{flakes,unit,examples,all_recommended}-{default}-{dev,pkg}
-build = wheel
```
Comment (ceball) [inline tox.ini]: I'd hope you could leave this; please let me know if things don't work with this still present.

**Comment by philippjfr** [comment]
You're right that it's a build time dependency only. Going back to the cleaner approach would be nice but I couldn't work out how to satisfy the nodejs dependency in that case.

File: .travis.yml (hunk 1 above)
Comment (philippjfr) [reply .travis.yml]: Build time dependency only

File: .travis.yml (hunk 2 above)
Comment (philippjfr) [reply .travis.yml]: No this was intentional, at some point the package upload started failing with some error about testpypi not being a valid argument.

File: tox.ini (hunk 4 above)
Comment (philippjfr) [reply tox.ini]: I'll revert this, I was trying to work around py27 issues and this was just collateral.

File: .travis.yml (hunk 2 above)
Comment (philippjfr) [reply .travis.yml]: [Here](https://travis-ci.org/pyviz/panel/jobs/442481612)'s the actual error

```
$ doit ecosystem=pip package_upload -u $PYPIUSER -p $PYPIPASS -pypi ${PYPI}
ERROR: Invalid parameter: "testpypi". Must be a command, task, or a target.
Type "doit help" to see available commands.
Type "doit list" to see available tasks.
```

Comment (philippjfr) [reply .travis.yml]: Ah I see my error, will fix it now.

**Comment by ceball** [comment]
Just to be clear, what I meant above is that I think you should be able to continue using the "pip on conda" approach, so don't let my comment that it's not ideal hold up merging if this is otherwise ready. 

(The downside is that the pip package testing will also occur with the package build time dependencies present, i.e. it won't mimic the setup of a typical pip user - nodejs will be present. But I think that's ok for now. I'll revisit later.)

**Comment by philippjfr** [comment]
Since bokeh 1.0 is now released I'll merge.

**Pull request merged**